packages/field-base/src/field-mixin.js

```diff
@@ -15,9 +17,10 @@ import { ValidateMixin } from './validate-mixin.js';
  * @polymerMixin
  * @mixes LabelMixin
  * @mixes ValidateMixin
+ * @mixes ElementMixin
```

Comment:
Let's avoid coupling `ElementMixin` with this mixin because it exists for a different purpose (collecting usage statistics at development time). We should consider moving controllers logic to `SlotMixin` instead.

Reply:
Do you mean to rather extend `SlotMixin` from `ElementMixin`?

Reply:
1. Do not extend `ElementMixin` in any other mixin, it's needed for components only.
2. Move `addController` API and tests to `SlotMixin` to not have it in `ElementMixin`

Reply:
Submitted a separate PR to move the controllers logic to the separate `ControllersMixin`: https://github.com/vaadin/web-components/pull/3069

Reply:
Changed `FieldMixin` such that it now extends `ControllerMixin` instead of `ElementMixin`.